lib/Serialization/SerializeSIL.cpp

```diff
@@ -392,10 +392,7 @@ void SILSerializer::writeSILFunction(const SILFunction &F, bool DeclOnly) {
   // Assign a value ID to each SILInstruction that has value and to each basic
   // block argument.
   unsigned ValueID = 0;
-  llvm::ReversePostOrderTraversal<SILFunction *> RPOT(
-      const_cast<SILFunction *>(&F));
-  for (auto Iter = RPOT.begin(), E = RPOT.end(); Iter != E; ++Iter) {
-    auto &BB = **Iter;
+  for (const SILBasicBlock &BB : F) {
```

Comment:
Fine. Another thing to consider here is that it is weird that you are using const SILBasicBlock & where in all other places in the commit you are using auto &. I don't feel that strongly about this though.